### PR TITLE
Add config system and some optimizations

### PR DESCRIPTION
**Pull request by VaibhavSys**

### CODE REVIEW

**Comment by VaibhavSys** [comment]
Config file could have been made better however, I wasn't able to categorize them.

**Comment by AceyDoCodes** [comment]
use pydantic for config its better to work with tbh

**Comment by VaibhavSys** [comment]
> use pydantic for config its better to work with tbh

I believe that something minimal and simple like TOML would be way better as there is no need of complexity in this.

File: src/exts/suggestion.py [hunk 1]
```diff
@@ -52,14 +53,15 @@ async def _suggest(
                 text=f"By {str(interaction.user)} (ID {interaction.user.id})"
             )
 
-        channel = self.bot.get_channel(831425425510760478)
+        channel = self.suggestion_channel
```
Comment (MaskDuck) [inline src/exts/suggestion.py]: suggestion_channel is an int. we cant interact with an int.

File: src/exts/bans.py [hunk 2]
```diff
@@ -58,8 +53,13 @@ def __init__(self):
         self.add_item(self.__why)
         self.add_item(self.__improvement)
 
+        self.MAIN_SERVER_ID = self.bot.config["main_server"]
+        self.APPEAL_SERVER_ID = self.bot.config["appeal_server"]
+        self.APPEAL_CHANNEL_ID = self.bot.config["appeal_channel"]
+        self.UNBANNED_CHANNEL_ID = self.bot.config["unbanned_channel"]
```
Comment (MaskDuck) [inline src/exts/bans.py]: typehint these with typing.Final[int]

File: src/__main__.py [hunk 3]
```diff
@@ -26,6 +27,9 @@ def __init__(self):
         self.load_extension("exts.stars")
         self.load_extension("exts.roles")
 
+    with open("config.toml", "r") as config_file:
+        config = parse(config_file.read())
```
Comment (MaskDuck) [inline src/__main__.py:31]:
```suggestion
        self.config = parse(config_file.read())
```

**Review by MaskDuck — CHANGES_REQUESTED** [review]

**Comment by AceyDoCodes** [comment]
> > use pydantic for config its better to work with tbh
> 
> I believe that something minimal and simple like TOML would be way better as there is no need of complexity in this.

its not that complicated :/ you just install pydantic subclass pydantic.BaseModel and add some shts and boom just use the class

**Comment by VaibhavSys** [comment]
> > > use pydantic for config its better to work with tbh
> > 
> > 
> > I believe that something minimal and simple like TOML would be way better as there is no need of complexity in this.
> 
> its not that complicated :/ you just install pydantic subclass pydantic.BaseModel and add some shts and boom just use the class

TOML is simple, just whats needed for *this*. Pydantic would be too complex for *this* I believe.

File: src/exts/suggestion.py (hunk 1 above)
Comment (VaibhavSys) [reply src/exts/suggestion.py]: [05e5d0a](https://github.com/MaskDuck/impostor/pull/13/commits/05e5d0af3faf9af68a830f127b0fd1c6a7da7782)

File: src/__main__.py (hunk 3 above)
Comment (VaibhavSys) [reply src/__main__.py:31]: The `config` dict is already on the `bot` object.

File: src/exts/bans.py [hunk 4]
```diff
@@ -58,8 +53,13 @@ def __init__(self):
         self.add_item(self.__why)
         self.add_item(self.__improvement)
 
+        self.MAIN_SERVER_ID: int = self.bot.config["main_server"]
+        self.APPEAL_SERVER_ID: int = self.bot.config["appeal_server"]
+        self.APPEAL_CHANNEL_ID: int = self.bot.config["appeal_channel"]
+        self.UNBANNED_CHANNEL_ID: int = self.bot.config["unbanned_channel"]
```
Comment (MaskDuck) [inline src/exts/bans.py]:
```suggestion
        self.MAIN_SERVER_ID: Final[int] = self.bot.config["main_server"]
        self.APPEAL_SERVER_ID: Final[int] = self.bot.config["appeal_server"]
        self.APPEAL_CHANNEL_ID: Final[int] = self.bot.config["appeal_channel"]
        self.UNBANNED_CHANNEL_ID: Final[int] = self.bot.config["unbanned_channel"]
```

File: src/exts/suggestion.py [hunk 5]
```diff
@@ -89,7 +91,7 @@ async def _deny(
             name="why", description="Why did you deny this request?", required=True
         ),
     ):
-        channel = self.bot.get_channel(831425425510760478)
+        channel = self.suggestion_channel
```
Comment (MaskDuck) [inline src/exts/suggestion.py]: i've already said, suggestion_channel is an int. we're interact with channel directly.

**Review by MaskDuck — CHANGES_REQUESTED** [review]
also it would be nice to use pydantic tbh.

File: src/__main__.py [hunk 6]
```diff
@@ -26,6 +27,9 @@ def __init__(self):
         self.load_extension("exts.stars")
         self.load_extension("exts.roles")
 
+    with open("config.toml", "r") as config_file:
+        self.config = parse(config_file.read())
```
Comment (MaskDuck) [inline src/__main__.py]:
```suggestion
        config = parse(config_file.read())
```

File: src/exts/suggestion.py (hunk 5 above)
Comment (VaibhavSys) [reply src/exts/suggestion.py]: then line 57 would be invalidiated and would have to get the channel again. Currently, it is going to get that channel and use the channel object so it should work.

Comment (MaskDuck) [reply src/exts/suggestion.py]:
```suggestion
        channel = interaction.guild.get_channel(self.suggestion_channel)
```

**Review by MaskDuck — APPROVED** [review]
lgtm

**Pull request merged**